include/session/config/groups/members.hpp

```diff
@@ -127,6 +129,17 @@ struct member {
         supplement = false;
     }
 
+    /// API: groups/member::invite_needs_send
+    ///
+    /// Returns whether the user needs an invite sent to them.  Returns true if so (whether or
+    /// not that invitation has been sent).
+    ///
+    /// Inputs: none
+    ///
+    /// Outputs:
+    /// - `bool` -- true if the user needs an invitation to be sent, false otherwise.
+    bool invite_needs_send() const { return invite_status == INVITE_NOT_SENT; }
```

Comment:
should this maybe be tweak to make it clear that it isn't returning true if the invite was already tried to be sent, but failed?

Reply:
Yea probably a good idea - I'll rename the functions to just be `invite_not_sent` and `promotion_not_sent` to avoid the ambiguity for now, I think down the track it'd be nicer to expose it as an enum instead of having individual functions but I wanted to avoid changing the behaviour too much at this stage